src/netaddress.cpp

```diff
@@ -235,7 +243,7 @@ bool CNetAddr::IsLocal() const
 
     // IPv6 loopback (::1/128)
     static const unsigned char pchLocal[16] = {0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1};
-    if (memcmp(ip, pchLocal, 16) == 0)
+    if (IsIPv6() && memcmp(ip, pchLocal, 16) == 0)
         return true;
```

Comment:
In general some of this might be optimized out by the compiler,  
And i would prefer for the transition period something like this that we can refactor out when all works fine
to an compiler and bytecode friendly version.

I would prefer constructions for all those ipv6 overrides in the rest of the code to look like this
suggested change: 
``` Code
if  (memcmp(ip, pchLocal, 16) == 0) && IsIPv6();
```
And some logging here and in the other overrides  i.e.  

``` Code
LogPrint(BCLog::ADDV2, "old isXXXX() says so, ADDRv2 m_net did override with  value = xxx .....   
```